pylint/checkers/strings.py

```diff
@@ -250,15 +292,15 @@ def visit_binop(self, node):
         except utils.UnsupportedFormatCharacter as e:
             c = format_string[e.index]
             self.add_message('bad-format-character',
-                             node=node, args=(c, ord(c), e.index))
+                             line=node.lineno, args=(c, ord(c), e.index))
```

Comment:
I don't think those modifications are necessary.

Reply:
They are, sadly:
> making StringFormatChecker inherit from BaseTokenChecker and implement ITokenChecker forces to change all the .add_message call sites in this class, because it now requires a line parameter:
https://github.com/PyCQA/pylint/blob/master/pylint/utils.py#L374
https://github.com/PyCQA/pylint/blob/master/pylint/utils.py#L153

Alternatively, I could move this new check from `StringFormatChecker` into `StringConstantChecker`, which would solve all those issues
I'm going to do so unless someone comment against it here

Reply:
Extracting your check to individual class is a great idea, go for it!